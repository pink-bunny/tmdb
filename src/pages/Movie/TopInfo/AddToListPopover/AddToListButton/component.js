import React from 'react';
import { Button } from 'antd';
import PropTypes from 'prop-types';

const AddToListButton = ({
  name,
  handleCLick
}) => (
  <Button
    type="link"
    onClick={handleCLick}
  >
    {name}
  </Button>
);

AddToListButton.propTypes = {
  name: PropTypes.string.isRequired,
  handleCLick: PropTypes.func.isRequired
};

export default AddToListButton;
