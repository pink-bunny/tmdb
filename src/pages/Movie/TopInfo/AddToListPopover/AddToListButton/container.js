import React from 'react';
import PropTypes from 'prop-types';

import AddToListButtonComponent from './component';

class AddToListButton extends React.Component {
  handleCLick = () => {
    const { id, addToList } = this.props;
    addToList(id);
  }

  render() {
    const { name } = this.props;

    return (
      <AddToListButtonComponent
        handleCLick={this.handleCLick}
        name={name}
      />
    );
  }
}

AddToListButton.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  addToList: PropTypes.func.isRequired
};

export default AddToListButton;
