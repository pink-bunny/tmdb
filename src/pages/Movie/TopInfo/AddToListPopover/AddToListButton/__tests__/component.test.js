import React from 'react';
import { shallow } from 'enzyme';

import AddToListButton from '../component';

describe('AddToListButton component', () => {
  const requiredProps = {
    name: 'Mock name',
    handleCLick: jest.fn()
  };
  const wrapper = shallow(<AddToListButton {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
