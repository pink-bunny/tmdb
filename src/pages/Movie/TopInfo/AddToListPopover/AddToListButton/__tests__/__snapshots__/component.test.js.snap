// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToListButton component matches snapshot with required props 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[MockFunction]}
  type="link"
>
  Mock name
</Button>
`;
